### PR TITLE
feat(dlm): support to connect postgres datasource

### PR DESCRIPTION
**Pull request by kiko-art**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
The connection of Postgres data source is supported.
#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/factory/OBConsoleDataSourceFactory.java [hunk 1]
```diff
@@ -313,6 +314,10 @@ public static String getDefaultSchema(@NonNull ConnectionConfig connectionConfig
                     return getSchema(defaultSchema, connectionConfig.getDialectType());
                 }
                 return getSchema(OdcConstants.MYSQL_DEFAULT_SCHEMA, connectionConfig.getDialectType());
+            case POSTGRE_SQL:
+                if (StringUtils.isNotEmpty(defaultSchema)) {
```
Comment (LuckyPickleZZ) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/factory/OBConsoleDataSourceFactory.java:318]: If the defaultSchema is empty ,this function would return `null`. Could this cause potential NPE?

File: server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresConnectionExtension.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return OdcConstants.POSTGRES_DRIVER_CLASS_NAME;
+    }
+
+    @Override
+    public List<ConnectionInitializer> getConnectionInitializers() {
+        return null;
```
Comment (yizhouxw) [inline server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresConnectionExtension.java:91]: avoid retun null, use `Collections.emptyList()` instead

File: server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresInformationExtension.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+@Extension
+public class PostgresInformationExtension implements InformationExtensionPoint {
+    @Override
+    public String getDBVersion(Connection connection) {
+        String querySql = "SELECT current_setting('server_version');";
+        String dbVersion;
+        try {
+            dbVersion = JdbcOperationsUtil.getJdbcOperations(connection).query(querySql, rs -> {
+                if (!rs.next()) {
+                    return null;
```
Comment (yizhouxw) [inline server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresInformationExtension.java:36]: avoid return null

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/processor/AbstractDlmPreprocessor.java [hunk 4]
```diff
@@ -130,11 +139,15 @@ public Map<DataArchiveTableConfig, String> getDataArchiveSqls(Database database,
...
         try {
             DialectType dbType = database.getDataSource().getDialectType();
-            if (!dbType.isOracle() && !dbType.isMysql()) {
+            if (!dbType.isOracle() && !dbType.isMysql() && !dbType.isPostgreSql()) {
                 throw new UnsupportedException();
             }
             SqlBuilder sqlBuilder = dbType.isMysql() ? new MySQLSqlBuilder() : new OracleSqlBuilder();
-            sqlBuilder.append("SELECT 1 FROM ").identifier(database.getName(), table.getTableName());
+            if (dbType.isPostgreSql()) {
+                sqlBuilder.append("SELECT 1 FROM ").identifier("public", table.getTableName());
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/processor/AbstractDlmPreprocessor.java:147]: why hardcode use 'public' schema for postgres?

File: server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresInformationExtension.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+@Extension
+public class PostgresInformationExtension implements InformationExtensionPoint {
+    @Override
+    public String getDBVersion(Connection connection) {
+        String querySql = "SELECT current_setting('server_version');";
+        String dbVersion;
+        try {
+            dbVersion = JdbcOperationsUtil.getJdbcOperations(connection).query(querySql, rs -> {
+                if (!rs.next()) {
+                    return "";
```
Comment (yizhouxw) [inline server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresInformationExtension.java:36]: is query version got empty result here an expected behavior?
if not, throw exception instead.


here you can simply use queryForObject method

File: server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresInformationExtension.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                }
+                return rs.getString(1);
+            });
+        } catch (Exception e) {
+            throw new BadRequestException(ErrorCodes.QueryDBVersionFailed,
+                    new Object[] {e.getMessage()}, e.getMessage());
+        }
+        if (dbVersion == null) {
```
Comment (yizhouxw) [inline server/plugins/connect-plugin-postgres/src/main/java/com/oceanbase/odc/plugin/connect/postgres/PostgresInformationExtension.java:44]: if queryForObject used, this validation logic can be removed

File: libs/db-browser/pom.xml [hunk 7]
```diff
@@ -99,6 +99,7 @@
         <license-maven-plugin.version>3.0</license-maven-plugin.version>
         <oracle.jdbc.version>21.1.0.0</oracle.jdbc.version>
         <caffeine.version>2.8.0</caffeine.version>
+        <postgres.jdbc.version>42.7.3</postgres.jdbc.version>
```
Comment (yhilmare) [inline libs/db-browser/pom.xml:102]: please upgrade the db-browser's version i this pom.xml and odc's pom

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+
+    protected JdbcOperations jdbcOperations;
+
+    public PostgresSchemaAccessor(@NonNull JdbcOperations jdbcOperations) {
+        this.jdbcOperations = jdbcOperations;
+    }
+
+    @Override
+    public List<String> showDatabases() {
+        String sql = "SELECT datname FROM pg_database;";
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java:63]: I don't think this should return database, schemas will be better~

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+        return jdbcOperations.queryForList(sql, String.class);
+    }
+
+    @Override
+    public DBDatabase getDatabase(String schemaName) {
+        throw new UnsupportedOperationException("Not supported yet");
+    }
+
+    @Override
+    public List<DBDatabase> listDatabases() {
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java:73]: same as above

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+    @Override
+    public List<DBObjectIdentity> listUsers() {
+        throw new UnsupportedOperationException("Not supported yet");
+    }
+
+    @Override
+    public List<String> showTables(String schemaName) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT table_name FROM information_schema.tables "
+                + "WHERE table_schema = 'public' AND table_type = 'BASE TABLE'");
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java:101]: why should we use 'public' as the schema?

File: server/plugins/task-plugin-postgres/pom.xml [hunk 11]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yhilmare) [inline server/plugins/task-plugin-postgres/pom.xml:1]: have we tested import/export for postgresql?

we'd better remove this empty task plugin if we do't support it actually~

**Review by yhilmare — COMMENTED** [review]
we'd better add a new field `database` at ConnectionConfig

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/processor/AbstractDlmPreprocessor.java (hunk 4 above)
Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/processor/AbstractDlmPreprocessor.java:147]: In the Postgres database, the public schema serves as the repository for user-defined data objects. In contrast, other schemas such as pg_catalog schema contain metadata related to the database system, and the information_schema holds information regarding the entire database. Here, as we anticipate presenting user-defined data objects, we employ the public schema.

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java (hunk 10 above)
Comment (kiko-art) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java:101]: In the Postgres database, the public schema serves as the repository for user-defined data objects. In contrast, other schemas such as pg_catalog schema contain metadata related to the database system, and the information_schema holds information regarding the entire database. Here, as we anticipate presenting user-defined data objects, we employ the public schema.

File: server/plugins/task-plugin-postgres/pom.xml (hunk 11 above)
Comment (kiko-art) [reply server/plugins/task-plugin-postgres/pom.xml]: this module has been removed

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/processor/AbstractDlmPreprocessor.java (hunk 4 above)
Comment (kiko-art) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/processor/AbstractDlmPreprocessor.java]: The hard-coded part has been removed.

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java (hunk 9 above)
Comment (kiko-art) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java:77]: Has been changed to list schema.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/model/ConnectionConfig.java [hunk 12]
```diff
@@ -122,6 +122,11 @@ public class ConnectionConfig
      */
     private UserRole userRole;
 
+    /**
+     * 该参数表示一个数据库实例
+     */
+    private String databaseName;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/model/ConnectionConfig.java:128]: The comment can explain that this field is currently used only for Postgres

**Review by guowl3 — DISMISSED** [review]
lgtm

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_2__alter_connect_connection.sql [hunk 13]
```diff
@@ -0,0 +1,4 @@
+--
+-- Add column `database_name` to `connect_connection` table
+--
+alter table `connect_connection` add column `database_name` varchar(255) DEFAULT NULL COMMENT 'Parameters which represent an instance of a database';
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_2__alter_connect_connection.sql:4]: better catalog_name than database_name,

catalog is a common concept in database area.

database_name may confused with other concept.


and postgres also use catalog_name in dictionary table.

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleControllerHist.java [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+import io.swagger.annotations.ApiOperation;
+
+/**
+ * @Author：tinker
+ * @Date: 2022/11/22 14:27
+ * @Descripition:
+ */
+
+@RestController
+@RequestMapping("/api/v2/schedule")
```
Comment (yizhouxw) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleControllerHist.java:56]: ScheduleConroller seems not related with PG archiver?

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_2__alter_connect_connection.sql (hunk 13 above)
Comment (kiko-art) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_2__alter_connect_connection.sql]: The databaseName field has been changed to catalogName

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleControllerHist.java (hunk 14 above)
Comment (kiko-art) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleControllerHist.java]: The latest dev/4.3.x code is merged.

**Review by guowl3 — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]
lgtm, congratuations!

**Review by PeachThinking — APPROVED** [review]

**Review by LuckyPickleZZ — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm
